lib/Conversion/StableHLOToTTIR/StableHLOToTTIRPatterns.cpp

```diff
@@ -321,6 +320,73 @@ class StableHLOToTTIRBroadcastInDimOpConversionPattern
   }
 };
 
+class StableHLOToTTIRConcatOpConversionPattern
+    : public OpConversionPattern<mlir::stablehlo::ConcatenateOp> {
+
+  using OpConversionPattern<
+      mlir::stablehlo::ConcatenateOp>::OpConversionPattern;
+
+public:
+  LogicalResult
+  matchAndRewrite(mlir::stablehlo::ConcatenateOp srcOp,
+                  mlir::stablehlo::ConcatenateOp::Adaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    // Check legality of the operation
+    LogicalResult err = checkBasicLegality(srcOp, adaptor, rewriter);
+    if (failed(err)) {
+      return err;
+    }
+
+    // Check that the dimension is valid
+    int32_t dimension = int32_t(adaptor.getDimension());
+    auto rankedTensorType =
+        mlir::dyn_cast<mlir::RankedTensorType>(srcOp.getOperand(0).getType());
+    if (dimension >= rankedTensorType.getRank()) {
+      return rewriter.notifyMatchFailure(srcOp,
+                                         "Invalid concatenation dimension.");
+    }
+
+    // // Create the output tensor type based on inputs
```

Comment:
`//` are written twice. Please remove one of them and fix it in other places as well.